### PR TITLE
Instantiate admins to native contracts

### PR DESCRIPTION
**Pull request by Buckram123**

### CODE REVIEW

**Review by cyberhoward — COMMENTED** [review]
I think I would prefer the admin to be required in the deploy_on function and not be optional in the Instantiate messages. 

It's still very easy to set the admin to the sender in that case of we want to, just by providing the address of the current sender in the deploy function. WDYT?

File: framework/contracts/native/account-factory/src/contract.rs [hunk 1]
```diff
@@ -33,8 +33,12 @@ pub fn instantiate(
     cw2::set_contract_version(deps.storage, ACCOUNT_FACTORY, CONTRACT_VERSION)?;
 
     CONFIG.save(deps.storage, &config)?;
-    // Set up the admin as the creator of the contract
-    cw_ownable::initialize_owner(deps.storage, deps.api, Some(info.sender.as_str()))?;
+    // Set up the admin
+    let owner = match msg.admin.as_deref() {
```
Comment (Kayanski) [inline framework/contracts/native/account-factory/src/contract.rs:37]: This code is repeated 4 times, could we have a function instead that initializes the owner base on a optional admin field ?

**Review by Kayanski — APPROVED** [review]

File: framework/contracts/native/account-factory/src/contract.rs (hunk 1 above)
Comment (Buckram123) [reply framework/contracts/native/account-factory/src/contract.rs]: True, thanks

**Comment by Buckram123** [comment]
> I think I would prefer the admin to be required in the deploy_on function and not be optional in the Instantiate messages.

Agree, `admin: None` could also mean no-admin, which is not in our case

**Pull request merged**